rotkehlchen/accounting/structures/balance.py

```diff
@@ -18,6 +20,67 @@ class BalanceType(DBEnumMixIn):
     LIABILITY = 2
 
 
+def _build_balances_with_assets_groups(
+        balances: dict['Asset', 'Balance'],
+        assets_to_collection: dict[str, int],
+) -> dict['Asset', dict[str, Any]]:
+    """
+    Given a list of balances group the assets that are multichain or form a group aggregating
+    their value.
+
+    Returns a mapping of asset to their balance. The structure adheres to the following example
+
+    {'eip155:1/erc20:0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48': {
+        'amount': '5',
+        'breakdown': {
+            'eip155:1/erc20:0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48': {
+                'amount': '2',
+                'usd_value': '2',
+            },
+            'eip155:10/erc20:0x7F5c764cBc14f9669B88837ca1490cCa17c31607': {
+                'amount': '3',
+                'usd_value': '3',
+            },
+        },
+        'usd_value': '5',
+    }}
+    """
+    assets_with_only_balance: dict['Asset', dict[str, str]] = {}
+    assets_with_breakdown: dict['Asset', Any] = {}
+    total_collections_balance: defaultdict[int, Balance] = defaultdict(Balance)
+    assets_with_collections: dict[int, dict[Asset, tuple[int, dict[str, str]]]] = defaultdict(dict)
+    for asset, balance in balances.items():
+        if asset not in assets_to_collection:
+            # if the asset doesn't belong to any collection just add it as a mapping to balance
+            assets_with_only_balance[asset] = balance.serialize()
+        else:
+            # aggregate the balances of the assets in the same collection and store the balances
+            # for each asset
+            collection = assets_to_collection[asset.identifier]
+            total_collections_balance[collection] += balance
+            # we need the chain id to later sort and know what assets in the set will be
+            # used to represent the others in the collection.
+            crypto_asset = asset.resolve_to_crypto_asset()
+            if isinstance(crypto_asset, EvmToken) is True:
+                sorting_key = crypto_asset.chain_id  # type: ignore  # noqa: E501 mypy fails to see that this could be EvmToken
+            else:
+                sorting_key = 0
```

Comment:
```suggestion
            else:
                # we need to do this to avoid issues in the case of collections for non evm tokens
                sorting_key = 0
```